packages/vite/src/node/build.ts

```diff
@@ -508,11 +514,14 @@ async function doBuild(
     }
 
     if (Array.isArray(outputs)) {
-      const res = []
-      for (const output of outputs) {
-        res.push(await generate(output))
-      }
-      return res
+      // the first output is for legacy bundle which should be excuted first
+      return [
+        await generate(outputs[0]),
```

Comment:
**question:** Is it possible that `outputs` have a `length` of `0`?  
If so we now execute `generate` with `undefined` :thinking:  

Previously it was not called do to that the for catched this case.

Reply:
At least I don't think this would break anything, cause `generate` allows to be called with `undefined` :thinking:

Reply:
even if don't have a  user `rollupOption.output`, there's always a legacy output prepended internally

Reply:
I see :thinking:  
I still want to await at least @patak-js response about this PR

Sorry that I make it so hard for such an "easy" PR, nothing personal :sweat_smile:

Reply:
We will discuss this tomorrow with Evan and find out if there were others/more reasons behind the sequential

Reply:
of course it should be careful and I don't want to inject trash code into this amazing work neither😂